core/src/main/scala/org/bitcoins/core/protocol/tlv/ValueIterator.scala

```diff
@@ -33,6 +34,11 @@ case class ValueIterator(value: ByteVector) {
     bytes
   }
 
+  /** Same as [[take]] but doesn't forward the iterator */
+  def takeNoSkip[E <: NetworkElement](factory: Factory[E]): E = {
```

Comment:
this should be called `peek` as I believe that is the convention with data structures like this